packages/perspective-viewer-hypergrid/test/js/selection.spec.js

```diff
@@ -0,0 +1,80 @@
+ */
+
+const utils = require("@finos/perspective-test");
+const path = require("path");
+
+utils.with_server({}, () => {
+    describe.page(
+        "selectable.html",
+        () => {
+            describe("no pivots", () => {
+                describe("selecting a row", () => {
+                    test.capture("highlights the row with no pivots", async page => {
+                        await page.$("perspective-viewer");
+                        await page.focus("perspective-viewer");
+                        await page.mouse.click(80, 60);
+                        await page.mouse.move(80, 40);
+                        await page.waitForSelector("perspective-viewer:not([updating])");
+                    });
+
+                    test.capture("keeps selection when row moves with no pivots", async page => {
+                        const viewer = await page.$("perspective-viewer");
+                        await page.evaluate(element => {
+                            element.setAttribute("sort", '[["number", "asc"]]');
+                        }, viewer);
+
+                        await page.waitForSelector("perspective-viewer:not([updating])");
+
+                        await page.focus("perspective-viewer");
+                        await page.mouse.click(80, 60);
+
+                        await page.evaluate(element => {
+                            element.update([{name: "Homer", number: 300}]);
+                        }, viewer);
+                        await page.waitForSelector("perspective-viewer:not([updating])");
+                    });
+                });
+            });
+            describe("row pivots", () => {
+                describe("selecting a row", () => {
+                    test.capture("highlights the row with row pivots", async page => {
+                        const viewer = await page.$("perspective-viewer");
+                        await page.evaluate(element => {
+                            element.setAttribute("row-pivots", '["name"]');
+                        }, viewer);
+                        await page.waitForSelector("perspective-viewer:not([updating])");
+
+                        await page.focus("perspective-viewer");
+                        await page.mouse.click(80, 60);
+                        await page.mouse.move(80, 40);
+                        await page.waitForSelector("perspective-viewer:not([updating])");
+                    });
+
+                    test.capture("keeps selection when row moves with row pivots", async page => {
+                        const viewer = await page.$("perspective-viewer");
+                        await page.evaluate(element => {
+                            element.setAttribute("sort", '[["number", "asc"]]');
+                            element.setAttribute("row-pivots", '["name"]');
+                        }, viewer);
+
+                        await page.focus("perspective-viewer");
+                        await page.mouse.click(80, 60);
```

Comment:
This test [flapped in Travis](https://travis-ci.org/finos/perspective/jobs/624263934#L1232) - maybe potentially because of the click race condition above?  We unfortunately have very bad state capture for these tests, especially on Travis.